src/languages/rust.ts

```diff
@@ -0,0 +1,49 @@
+import {
+  createPatternMatchers,
+  argumentMatcher,
+  leadingMatcher,
+  conditionMatcher,
+  trailingMatcher,
+} from "../util/nodeMatchers";
+import { NodeMatcherAlternative, ScopeType } from "../typings/Types";
+
+const STATEMENT_TYPES = [
+  "use_declaration",
+  "while_expression",
+  "loop_expression",
+  "block",
```

Comment:
You probably don't want to include `"block"`, as that will trap you between the curly brackets if you say "take state" within an if statement body